src/module_library/von_caemmerer_c4_assimilation.h

```diff
@@ -0,0 +1,142 @@
+
+#ifndef VON_CAEMMERER_C4_H
+#define VON_CAEMMERER_C4_H
+
+// standard library includes
+#include <cmath>
+#include <algorithm>
+#include <array>
+#include <limits>
+
+// includes from biocro
+#include "../framework/quadratic_root.h"
```

Comment:
It's good practice to only use `include` directives where they are required. In this case, `quadratic_root_minus` from `quadratic_root.h` is used in `von_caemmerer_c4_assimilation.cpp`, so the `include` statement should go in that file. Otherwise the `.cpp` file is relying on the `.h` file to include `quadratic_root.h` for it.

This particular case is not very troublesome, but it's possible to end up with complicated chains of `include` statements. Then if you remove an `include` from one from a file that no longer needs it, suddenly there are compilation errors with other unrelated files. So it's better to just put them only where they are needed.